pkg/network/tracer/tracer_linux_test.go

```diff
@@ -1366,32 +1364,28 @@ func iptablesWrapper(t *testing.T, f func()) {
 	// Init iptables rule to simulate packet loss
 	rule := "INPUT --source 127.0.0.1 -j DROP"
 	create := strings.Fields(fmt.Sprintf("-I %s", rule))
-	remove := strings.Fields(fmt.Sprintf("-D %s", rule))
 
+	state := testutil.IptablesSave(t)
 	createCmd := exec.Command(iptables, create...)
 	err = createCmd.Start()
 	assert.Nil(t, err)
 	err = createCmd.Wait()
 	assert.Nil(t, err)
 
 	defer func() {
-		// Remove the iptable rule
-		removeCmd := exec.Command(iptables, remove...)
-		err = removeCmd.Start()
-		assert.Nil(t, err)
-		err = removeCmd.Wait()
-		assert.Nil(t, err)
+		testutil.IptablesRestore(t, state)
 	}()
 
 	f()
 }
 
-func setupDNAT(t *testing.T) {
+func setupDNAT(t *testing.T) []byte {
```

Comment:
Should this be using `t.Cleanup(teardownDNAT...)` internally, rather than exposing the state?

Reply:
Yes, changed